### PR TITLE
Update developer docs

### PR DESCRIPTION
**Pull request by zpatrick**
closes https://github.com/quintilesims/layer0/issues/44

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,18 +2,195 @@
...
+| Function      | Command       |
+| ------------- |:--------------|
+| Run program        | `go run main.go` |
+| Build binary       | `go build -o l0-runner` |
+| Build Docker Image | `make build` |
+| Release            | `make release` |
+
+## Docs
+The documentation for Layer0 exists in `layer0/docs-src`. 
+We use [mkdocs](http://www.mkdocs.org/) to compile our markdown docs into html and css.
```
Comment (jonathanstallings) [inline README.md]: We currently require `mkdocs` `>v0.16.0`. Do we need to make that explicit here? It shouldn't be a problem if a new dev uses `/docs-src/make deps` (installing latest), but a system with an older pre-installed version of `mkdocs` could run into a problem.

Comment (zpatrick) [inline README.md]: yep, added

File: README.md [hunk 2]
```diff
@@ -2,18 +2,195 @@
...
+
+#### Test your Changes
+Once you have make changes to your code, you should run all of the unit tests:
+```
+make unittest
+```
+
+Once the unit tests are passing, you should run the smoke tests and system tests. 
+However, these tests require that your update changes are running in AWS. 
+This requires bubilding a new docker image for the API and Runner, pushing them to Dockerhub, and updating your Layer0 instance to run the new images.
```
Comment (zpatrick) [inline README.md]: fixed

File: README.md [hunk 3]
```diff
@@ -2,18 +2,195 @@
...
+## Layer0 Setup
+The Layer0 Setup tool (commonly called `l0-setup`) is used to provision and manage resources in AWS that a Layer0 instance needs to run correctly. 
 
-* Layer0 API source is located in `/api`. For developer notes, see [/api/README.md](/api/README.md).
-* Layer0 Setup source (`l0-setup`) is located in `/setup`. For developer notes, see [/setup/README.md](/setup/README.md).
-* Layer0 CLI source (`l0`) is located in `/cli`. For developer notes, see [/cli/README.md](/cli/README.md).
-* Layer0 Runner source is located in `/runner`. For developer notes, see [/runner/README.md](/runner/README.md).
-* Layer0 Documentation is located in `/docs`. For developer notes, see [/docs/README.md](/docs/README.md).
+#### Common Functions
+*The following commands should be ran from the `layer0/setup` directory*
```
Comment (zpatrick) [inline README.md]: fixed

File: README.md [hunk 4]
```diff
@@ -2,18 +2,198 @@
...
+# Development Workflow
+A common workflow for working with Layer0
+
+####  Install a Layer0 Instance
+This is only required if you don't already have a Layer0 instance.
+We will use the Layer0 instances to test our changes AWS.
+
+```
+# install a new Layer0
+$ go run setup/main.go apply <instance>
```
Comment (jonathanstallings) [inline README.md]: Does this work? If run from `/layer0` directory, it appears that the apply action looks for templates in `/layer0` directory and faies. Should we direct developers to run from `/layer0/setup` directory instead?

Comment (zpatrick) [inline README.md]: ah, good catch. i'll fix that

**Pull request merged**